### DISCUSSION TITLE
Deploy from selected branches only

### DISCUSSION THREAD

**Discussion by dollybenson** [vercel/community#3298] [Help]
### Summary

Currently, vercel is deploying to preview whenever I push to any branch. I only want deployments for `main` and `preview` branches. `main` for production and `preview` for preview and development. I want vercel to ignore other branches. How do I do that?

### Steps to Reproduce

- Create a github repo with multiple branches
- Connect the main branch to vercel
- Push to any/all branches

**Comment by swarnava** [comment]
Hi,

You can take leverage of [Ignored Build Step](https://vercel.com/docs/concepts/projects/overview#ignored-build-step) feature to exclude branch using custom script.
